src/evaluators/BinaryExpression.js

```diff
@@ -57,14 +57,18 @@ export function getPureBinaryOperationResultType(
   lloc: ?BabelNodeSourceLocation,
   rloc: ?BabelNodeSourceLocation
 ): typeof Value {
-  function leakOrReportErrorIfNotPure(purityTest: (Realm, Value) => boolean, typeIfPure: typeof Value): typeof Value {
+  function havocIfPureAndReportErrorIfNot(
+    purityTest: (Realm, Value) => boolean,
+    typeIfPure: typeof Value
+  ): typeof Value {
     let leftPure = purityTest(realm, lval);
     let rightPure = purityTest(realm, rval);
     if (leftPure && rightPure) return typeIfPure;
 
     if (realm.isInPureScope()) {
-      if (!leftPure) Leak.leakValue(realm, lval);
-      if (!rightPure) Leak.leakValue(realm, rval);
+      // Allow unknown side effects on lval/rval, but assume no other side effects will occur
+      if (!leftPure) Havoc.value(realm, lval);
+      if (!rightPure) Havoc.value(realm, rval);
```

Comment:
Is `Havoc.value` clear? Maybe it could be `Havoc.add` or something that describes the verb for the action?

Reply:
Havoc is already a verb and Havoc.value can be read as "havoc this value". I don't think it can get any clearer than that.